src/main/java/com/pingcap/tikv/codec/DecimalUtils.java

```diff
@@ -30,18 +30,26 @@
      * @param cdi cdi is source data.
      * */
     public static double readDecimalFully(CodecDataInput cdi) {
-        if (cdi.size() < 3) {
+        if (cdi.available() < 3) {
             throw new IllegalArgumentException("insufficient bytes to read value");
         }
+
+        MyDecimal dec = new MyDecimal();
+        List<Integer> data = new ArrayList<>();
```

Comment:
I think I already did that way. You could scroll down a little bit and you would find out that `Ints.toArray`.